seqsero2/1.2.1/Dockerfile

```diff
@@ -25,48 +28,96 @@ LABEL maintainer3.email="kelsey.florek@slh.wisc.edu"
+ tar -xzf ${SALMID_VER}.tar.gz && \
+ rm -rvf ${SALMID_VER}.tar.gz
 
-# Install SPAdes 3.9.0
-RUN wget https://github.com/ablab/spades/releases/download/v3.9.0/SPAdes-3.9.0-Linux.tar.gz && \
- tar -xzf SPAdes-3.9.0-Linux.tar.gz && \
- rm -rf SPAdes-3.9.0-Linux.tar.gz
+# install SPAdes binary
+RUN wget http://cab.spbu.ru/files/release${SPADES_VER}/SPAdes-${SPADES_VER}-Linux.tar.gz && \
+  tar -xzf SPAdes-${SPADES_VER}-Linux.tar.gz && \
+  rm -rv SPAdes-${SPADES_VER}-Linux.tar.gz
 
 # Install SeqSero2; make /data
 RUN wget https://github.com/denglab/SeqSero2/archive/v${SEQSERO2_VER}.tar.gz && \
  tar -xzf v${SEQSERO2_VER}.tar.gz && \
- rm -rf v${SEQSERO2_VER}.tar.gz && \
+ rm -vrf v${SEQSERO2_VER}.tar.gz && \
  cd /SeqSero2-${SEQSERO2_VER}/ && \
  python3 -m pip install . && \
  mkdir /data
 
 # PATH edited to include /SeqSero2-1.1.0/bin in previous python3 command
-ENV PATH="${PATH}:/SPAdes-3.9.0-Linux/bin:/SalmID-0.122:/samtools-1.9" \
+ENV PATH="${PATH}:/SPAdes-${SPADES_VER}-Linux/bin:/SalmID-${SALMID_VER}:/samtools-${SAMTOOLS_VER}" \
     LC_ALL=C
 
 WORKDIR /data
+
+FROM app as test
+
+WORKDIR /test
+
+# install ncbi datasets tool (pre-compiled binary); place in $PATH
+RUN wget https://ftp.ncbi.nlm.nih.gov/pub/datasets/command-line/LATEST/linux-amd64/datasets && \
+ chmod +x datasets && \
+ mv -v datasets /usr/local/bin
+
+# download an example assembly; test with SeqSero2
+# Salmonella enterica serovar Infantis genome: https://www.ncbi.nlm.nih.gov/data-hub/genome/GCA_007765495.1/
+# BioSample:SAMN07684583
+ARG GENBANK_ACCESSION="GCA_007765495.1"
+RUN datasets download genome accession ${GENBANK_ACCESSION} --filename ${GENBANK_ACCESSION}.zip && \
+    mkdir -v ${GENBANK_ACCESSION}-download && \
+    unzip ${GENBANK_ACCESSION}.zip -d ${GENBANK_ACCESSION}-download && \
+    rm ${GENBANK_ACCESSION}.zip && \
+    mv -v ${GENBANK_ACCESSION}-download/ncbi_dataset/data/${GENBANK_ACCESSION}/${GENBANK_ACCESSION}*.fna ${GENBANK_ACCESSION}-download/ncbi_dataset/data/${GENBANK_ACCESSION}/${GENBANK_ACCESSION}.genomic.fna && \
+    SeqSero2_package.py \
+      -i ${GENBANK_ACCESSION}-download/ncbi_dataset/data/${GENBANK_ACCESSION}/${GENBANK_ACCESSION}.genomic.fna \
+      -t 4 \
+      -m k \
+      -d ${GENBANK_ACCESSION}-seqsero2-assembly-kmer-mode \
+      -n ${GENBANK_ACCESSION} \
+      -p 2 && \
+    grep 'Infantis' ${GENBANK_ACCESSION}-seqsero2-assembly-kmer-mode/SeqSero_result.txt
+
+# testing reads as input for the same Salmonella isolate
+# specifically the "allele" mode which does micro assembly first using SPAdes
+RUN wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR608/003/SRR6082043/SRR6082043_1.fastq.gz && \
+    wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR608/003/SRR6082043/SRR6082043_2.fastq.gz && \
+    SeqSero2_package.py \
+      -i SRR6082043_1.fastq.gz SRR6082043_2.fastq.gz \
+      -t 2 \
+      -m a \
+      -d ${GENBANK_ACCESSION}-seqsero2-reads-allele-mode \
+      -n ${GENBANK_ACCESSION} \
```

Comment:
I was a little confused at this test because you're still using the GenBank Accession for the name of your sample, but otherwise this is fine.

Is there any reason to continue to use the GENBANK_ACCESSION variable?

```
#15 38.50 2022-11-17 21:07:15 (4.33 MB/s) - 'SRR6082043_2.fastq.gz' saved [126288177]
#15 38.50 
#15 111.1 [bam_sort_core] merging from 0 files and 2 in-memory blocks...
#15 120.8 building database...
#15 120.8 mapping...
#15 120.8 check samtools version: 1.9
#15 120.8 assembling...
#15 120.8 blasting... 
#15 120.8 
#15 120.8 Sample name:	GCA_007765495.1
#15 120.8 Output directory:	/test/GCA_007765495.1-seqsero2-reads-allele-mode
#15 120.8 Input files:	/test/SRR6082043_1.fastq.gz	/test/SRR6082043_2.fastq.gz
#15 120.8 O antigen prediction:	7
#15 120.8 H1 antigen prediction(fliC):	r
#15 120.8 H2 antigen prediction(fljB):	1,5
#15 120.8 Predicted identification:	Salmonella enterica subspecies enterica (subspecies I)
#15 120.8 Predicted antigenic profile:	7:r:1,5
#15 120.8 Predicted serotype:	Infantis
#15 120.8 Note:	
#15 120.8 
#15 120.8 Predicted serotype:	Infantis
```

Reply:
I just left it because it's the same BioSample, just using the reads instead of the assembly as input. I can change it to reflect the SRR accession instead if that makes it more clear/understandable